core/src/main/java/org/testcontainers/utility/DockerImageName.java

```diff
@@ -103,11 +104,26 @@ public DockerImageName(String nameWithoutTag, @NotNull String version) {
 
         if (version.startsWith("sha256:")) {
             repo = remoteName;
-            versioning = new Sha256Versioning(version.replace("sha256:", ""));
+            versioning = new Versioning.Sha256Versioning(version.replace("sha256:", ""));
         } else {
             repo = remoteName;
-            versioning = new TagVersioning(version);
+            versioning = new Versioning.TagVersioning(version);
         }
+
+        compatibleSubstituteFor = null;
+    }
+
+    private DockerImageName(String rawName,
+                            String registry,
+                            String repo,
+                            @Nullable Versioning versioning,
```

Comment:
marked as `@Nullable` while the field isn't

Reply:
Fixed